src/Component/Rule/NameField/NameField.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { Input } from 'antd';
+import FormItem from 'antd/lib/form/FormItem';
+
+// default props
+interface DefaultNameFieldProps {
+  label: string;
+  placeholder: string;
+}
+// non default props
+interface NameFieldProps extends Partial<DefaultNameFieldProps> {
+  onChange: ((newValue: string) => void);
+}
+
+/**
+ * Input field for the rule name.
+ */
+class NameField extends React.Component<NameFieldProps, any> {
```

Comment:
Have you thought about using `React.StatelessComponent<NameFieldProps>` for components without state?
I'm not sure about the benefits but as it exists it may have some. :smile:

Reply:
Since this would cause bigger re-writes in the component definitions I suggest we postpone this until we are sure we have benefit from using ``StatelessComponent``.